crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -87,8 +90,15 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), anyhow::Error> {
         options.set_transport(rumqttc::Transport::tls_with_config(tls_config.into()));
     }
 
+    // the default keepalive in rumqttc is 60 sec
+    if !cmd.window.is_zero() && cmd.window.as_secs() < 60 {
+        options.set_keep_alive(cmd.window);
```

Comment:
Smart

Reply:
I'm okay with that. But be aware that the precision is 60 sec if the timeout is greater than 60s.

An option, is to set the keep alive to 5 seconds to have a more precise timeout.

Reply:
If user gives 1m10s, then indeed with 60s keepalive, the command will break after 2m.
On the other hand, if user provides 1h, every 5s ping sounds a bit too much.

Does it make simplify if we set the same keepalive value as provided unless it's smaller than the maximum value?
> the maximum Keep Alive interval is 18 hours, 12 minutes, and 15 seconds. (https://www.hivemq.com/blog/mqtt-essentials-part-10-alive-client-take-over/)

Reply:
> if user provides 1h, every 5s ping sounds a bit too much.

I would not bother too much with that. This command being used locally and for test purposes.

Reply:
Changed in 20884905d1e22b2e6db220a89fc9b773abd8c334